server/src/test/java/org/elasticsearch/action/search/SearchPhaseControllerTests.java

```diff
@@ -931,19 +935,36 @@ public void onFinalReduce(List<SearchShard> shards, TotalHits totalHits, Interna
         }
     }
 
-    public void testPartialMergeFailure() throws InterruptedException {
+    public void testPartialReduce() throws Exception {
+        for (int i = 0; i < 10; i++) {
```

Comment:
Are these loops just to make sure we get a whole bunch of random cases every run?

Reply:
yes,  mainly to exercice the batch reduced size